constants/const.go

```diff
@@ -0,0 +1,27 @@
+package constants
+
+const (
+	OSExitSuccess = 0
+	OSExitError   = -1
+)
+
+type OutMode int64
```

Comment:
Why is it int64, if the values go only from 1 to 4?

Reply:
This inconsistency with the int64 type comes from the original code and was chosen based on considerations for the architecture that will run the code. When picking a type, the register size was taken into account. On the other hand, the int type was chosen just because it was easy at the time. I agree that it would be better to make the two consistent. Using a 32 or 64 bit int would likely be best for performance, whereas using an 8 bit int may not be ideal. I have not verified how the Go compiler stores bytes in memory, but GCC typically pads 16 and 8 bit integer types to 32 or 64 bits, unless instructed otherwise. While I used to be sensitive to these low-level considerations, I have come to understand that it is better to leave them to the compiler unless there is a serious issue to address. In this case, the only problem I see is the inconsistent types used for basically the same enum.

Reply:
In the case of individual variables IMHO the size/padding is irrelevant, from a performance PoV, compared to the readability. I think it should be the same type as for the output type consts, because otherwise the reader will wonder what the difference means, even though it doesn't really mean anything.

Reply:
I've opened an issue for it - I think it's better to get this PR to finally merge, and change it in a new PR. Therefore I'm marking this conversation as resolved.

Reply:
https://github.com/AC-FuSa-Tools/nav/issues/27